### PR TITLE
[3.4.2] DTLS_RETRANSMISSION_TIMEOUT

### PR DESCRIPTION
**Pull request by nickAS21**
## Pull Request description

https://github.com/eclipse/leshan/issues/1002

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] PR name contains fix version. For example, "[3.3.4] Hotfix of some UI component" or "[3.4] New Super Feature".
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Required for internal contributors only.

## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.

### CODE REVIEW

File: application/src/main/resources/thingsboard.yml [hunk 1]
```diff
@@ -710,6 +710,8 @@ transport:
     dtls:
       # Enable/disable DTLS 1.2 support
       enabled: "${COAP_DTLS_ENABLED:false}"
+     # RFC7925_RETRANSMISSION_TIMEOUT_IN_MILLISECONDS = 9000
+      retransmission_timeout_in_sec: "${DTLS_RETRANSMISSION_TIMEOUT:9}"
```
Comment (smatvienko-tb) [inline application/src/main/resources/thingsboard.yml:714]: COAP_ prefix

Comment (smatvienko-tb) [reply application/src/main/resources/thingsboard.yml:714]: milliseconds

File: common/coap-server/src/main/java/org/thingsboard/server/coapserver/TbCoapDtlsSettings.java [hunk 2]
```diff
@@ -51,6 +52,9 @@ public class TbCoapDtlsSettings {
     @Value("${transport.coap.dtls.bind_port}")
     private Integer port;
 
+    @Value("${transport.coap.dtls.retransmission_timeout_in_sec}")
```
Comment (smatvienko-tb) [inline common/coap-server/src/main/java/org/thingsboard/server/coapserver/TbCoapDtlsSettings.java:55]: millis

File: common/coap-server/src/main/java/org/thingsboard/server/coapserver/TbCoapDtlsSettings.java [hunk 3]
```diff
@@ -51,6 +52,9 @@ public class TbCoapDtlsSettings {
...
+    private Integer dtlsRetransmissionTimeoutInSec;
```
Comment (smatvienko-tb) [inline common/coap-server/src/main/java/org/thingsboard/server/coapserver/TbCoapDtlsSettings.java:56]: change to primitive

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/config/LwM2MTransportServerConfig.java [hunk 4]
```diff
@@ -37,6 +37,10 @@
 @ConfigurationProperties(prefix = "transport.lwm2m")
 public class LwM2MTransportServerConfig implements LwM2MSecureServerConfig {
 
+    @Getter
+    @Value("${transport.coap.dtls.retransmission_timeout_in_sec}")
+    private Integer dtlsRetransmissionTimeoutInSec;
+
     @Getter
     @Value("${transport.lwm2m.timeout:}")
```
Comment (smatvienko-tb) [inline common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/config/LwM2MTransportServerConfig.java:45]: : non sense

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/DefaultLwM2mTransportService.java [hunk 5]
```diff
@@ -132,6 +133,7 @@ private LeshanServer getLhServer() {
         DtlsConnectorConfig.Builder dtlsConfig = new DtlsConnectorConfig.Builder(getCoapConfig(config.getPort(), config.getSecurePort(), config));
 
         dtlsConfig.set(DtlsConfig.DTLS_ROLE, DtlsConfig.DtlsRole.SERVER_ONLY);
+        dtlsConfig.set(DtlsConfig.DTLS_RETRANSMISSION_TIMEOUT, config.getDtlsRetransmissionTimeoutInSec(), TimeUnit.SECONDS);
```
Comment (smatvienko-tb) [inline common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/DefaultLwM2mTransportService.java:136]: all static or all non-static imports

File: transport/coap/src/main/resources/tb-coap-transport.yml [hunk 6]
```diff
@@ -97,6 +97,8 @@ transport:
     dtls:
       # Enable/disable DTLS 1.2 support
       enabled: "${COAP_DTLS_ENABLED:false}"
+      # RFC7925_RETRANSMISSION_TIMEOUT_IN_MILLISECONDS = 9000
```
Comment (smatvienko-tb) [inline transport/coap/src/main/resources/tb-coap-transport.yml:100]: make the same changes as thingsboard.yml

File: transport/lwm2m/src/main/resources/tb-lwm2m-transport.yml [hunk 7]
```diff
@@ -103,6 +103,10 @@ transport:
     enabled: "${TB_TRANSPORT_LOG_ENABLED:true}"
     max_length: "${TB_TRANSPORT_LOG_MAX_LENGTH:1024}"
   # Local LwM2M transport parameters
+  coap:
```
Comment (smatvienko-tb) [inline transport/lwm2m/src/main/resources/tb-lwm2m-transport.yml:106]: make the same changes as thingsboard.yml

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/DefaultLwM2mTransportService.java (hunk 5 above)
Comment (smatvienko-tb) [reply common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/DefaultLwM2mTransportService.java:136]: millis

**Review by smatvienko-tb — CHANGES_REQUESTED** [review]
please make changes as mentioned in comments

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/config/LwM2MTransportServerConfig.java (hunk 4 above)
Comment (nickAS21) [reply common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/config/LwM2MTransportServerConfig.java:45]: This is not the subject of my PR

File: common/coap-server/src/main/java/org/thingsboard/server/coapserver/TbCoapDtlsSettings.java (hunk 3 above)
Comment (nickAS21) [reply common/coap-server/src/main/java/org/thingsboard/server/coapserver/TbCoapDtlsSettings.java:56]: Ok

File: common/coap-server/src/main/java/org/thingsboard/server/coapserver/TbCoapDtlsSettings.java (hunk 2 above)
Comment (nickAS21) [reply common/coap-server/src/main/java/org/thingsboard/server/coapserver/TbCoapDtlsSettings.java:55]: Ok

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/DefaultLwM2mTransportService.java (hunk 5 above)
Comment (nickAS21) [reply common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/DefaultLwM2mTransportService.java:136]: Ok

File: transport/lwm2m/src/main/resources/tb-lwm2m-transport.yml (hunk 7 above)
Comment (nickAS21) [reply transport/lwm2m/src/main/resources/tb-lwm2m-transport.yml:106]: Ok

File: application/src/main/resources/thingsboard.yml (hunk 1 above)
Comment (nickAS21) [reply application/src/main/resources/thingsboard.yml:714]: Ok

**Pull request closed without merging**